datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -42,7 +42,7 @@ pub fn binary_operator_data_type(
     let result_type = if !any_decimal(lhs_type, rhs_type) {
```

Comment:
I als find `binary_oprator_data_type` to be confusing and redundant with `get_result_type` but we can fix that in a follow on PR perhaps

Reply:
Great suggestion, I will do in following PR